Description: Fix examples build with not all features enabled
Author: Arnaud Ferraris <aferraris@debian.org>
Forwarded: not-needed
Last-Update: 2025-10-01
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/examples/all-types.rs
+++ b/examples/all-types.rs
@@ -3,8 +3,10 @@
 use heed::byteorder::BE;
 use heed::types::*;
 use heed::{Database, EnvOpenOptions};
+#[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+#[cfg(all(feature = "serde-bincode", feature = "serde-json", feature = "serde"))]
 fn main() -> Result<(), Box<dyn Error>> {
     let path = tempfile::tempdir()?;
 
@@ -105,3 +107,7 @@
 
     Ok(())
 }
+
+#[cfg(not(all(feature = "serde-bincode", feature = "serde-json", feature = "serde")))]
+fn main() {
+}
--- a/examples/rmp-serde.rs
+++ b/examples/rmp-serde.rs
@@ -2,8 +2,10 @@
 
 use heed::types::{SerdeRmp, Str};
 use heed::{Database, EnvOpenOptions};
+#[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+#[cfg(all(feature = "serde-rmp", feature = "serde"))]
 fn main() -> Result<(), Box<dyn Error>> {
     let path = tempfile::tempdir()?;
 
@@ -34,3 +36,7 @@
 
     Ok(())
 }
+
+#[cfg(not(all(feature = "serde-rmp", feature = "serde")))]
+fn main() {
+}
